packages/@aws-cdk/aws-apigatewayv2/lib/common/domain-name.ts

```diff
@@ -55,10 +78,79 @@ export interface DomainNameProps {
    * The custom domain name
    */
   readonly domainName: string;
+
   /**
-   * The ACM certificate for this domain name
+   * DomainNameConfigurations for a domain name.
+   * @link https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-domainname.html#cfn-apigatewayv2-domainname-domainnameconfigurations
+   */
+  readonly domainNameConfigurations: DomainNameConfiguration[];
+
+  /**
+    * The mutual TLS authentication configuration for a custom domain name.
+    * @default - mTLS is not configured.
+    */
+  readonly mtls?: MTLSConfig;
+}
+
+/**
+ * Specifies the configuration for a an API's domain name.
+ */
+export interface DomainNameConfiguration {
+  /**
+   * The reference to an AWS-managed certificate for use by the edge-optimized
+   * endpoint for the domain name. For "EDGE" domain names, the certificate
+   * needs to be in the US East (N. Virginia) region.
    */
   readonly certificate: ICertificate;
+
+  /**
+   * The user-friendly name of the certificate that will be used by the endpoint for this domain name.
+   * @default null
+   */
+  readonly certificateName?: string;
+
+  /**
+    * The type of endpoint for this DomainName.
+    * @default REGIONAL
+    */
+  readonly endpointType?: EndpointType;
+
+  /**
+    * The ARN of the public certificate issued by ACM to validate ownership of your
+    * custom domain. Only required when configuring mutual TLS and using an ACM
+    * imported or private CA certificate ARN as the RegionalCertificateArn.
+    * @default null
+    */
+  readonly ownershipVerificationCertificate?: ICertificate;
+
+  /**
+    * The Transport Layer Security (TLS) version + cipher suite for this domain name.
+    * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-domainname.html
+    * @default SecurityPolicy.TLS_1_0
+    */
+  readonly securityPolicy?: SecurityPolicy;
+}
+
+/**
+ * The mTLS authentication configuration for a custom domain name.
+ */
+export interface MTLSConfig {
+  /**
+   * The bucket that the trust store is hosted in.
+   */
+  readonly bucket: IBucket;
+
+  /**
+   * The key in S3 to look at for the trust store.
+   */
+  readonly key: string;
+
+  /**
+   *  The version of the S3 object that contains your truststore.
+   *  To specify a version, you must have versioning enabled for the S3 bucket.
+   *  @default - latest version
+   */
+  readonly version?: string;
```

Comment:
Can you use same naming for mutualTLSCOnfiguration here including interface name?
https://docs.aws.amazon.com/cli/latest/reference/apigatewayv2/create-domain-name.html

Reply:
These are consistent with the v1 domain name code. Please refer to this https://github.com/aws/aws-cdk/blob/ca9320bbc22934a0f708364a6426f3977ea67f5a/packages/%40aws-cdk/aws-apigateway/lib/domain-name.ts#L187

If you still think I should change these names, I am fine with that. I don't have a strong preference. I just wanted the names to be consistent across v1 and v2, that's why I used these names.